File: 백준/Gold/1377. 버블 소트/버블 소트.java
import java.io.*;
import java.util.*;

public class Main {
    static int n;
    static int[][] graph; 
    static boolean[] visited;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int[][] arr = new int[n][2];

        for(int i=0;i<n;i++){
            int num = Integer.parseInt(br.readLine());
            arr[i][0] = num;
            arr[i][1] = i;
        }

        int maxN = 0;
        Arrays.sort(arr, Comparator.comparingInt(a -> a[0]));

        for(int i=0;i<n;i++){
            int diff = arr[i][1] - i;
            if(maxN<diff){
                maxN = diff;
            }
        }
        System.out.println(maxN+1);
    }
}

File: 백준/Gold/1377. 버블 소트/버블 소트.py
import sys
input = sys.stdin.readline

n = int(input())
arr = []

for i in range(n):
    arr.append((int(input()),i))
maxn = 0
arr = sorted(arr)
for i in range(n):
    if maxn<arr[i][1]-i:
        maxn = arr[i][1]-i
print(maxn+1)